src/rpc/methods/eth/trace.rs

```diff
@@ -0,0 +1,624 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::types::{
+    EthAddress, EthBlockTrace, EthBytes, EthCallTraceAction, TraceAction, TraceResult,
+};
+use super::utils::{decode_params, decode_return};
+use super::{
+    decode_payload, encode_filecoin_params_as_abi, encode_filecoin_returns_as_abi,
+    EthCallTraceResult, EthCreateTraceAction, EthCreateTraceResult,
+};
+use crate::eth::{EAMMethod, EVMMethod};
+use crate::rpc::methods::eth::lookup_eth_address;
+use crate::rpc::methods::state::ExecutionTrace;
+use crate::rpc::state::ActorTrace;
+use crate::shim::{actors::is_evm_actor, address::Address, error::ExitCode, state_tree::StateTree};
+use fil_actor_eam_state::v12 as eam12;
+use fil_actor_evm_state::v15 as evm12;
+use fil_actor_init_state::v12::ExecReturn;
+use fil_actor_init_state::v15::Method as InitMethod;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_shared4::{error::ExitCode as ExitCodeV4, METHOD_CONSTRUCTOR};
```

Comment:
Do we have to lock the actor version and the FVM version here? It would be great to avoid it and work through actor interface or shims.

Reply:
I can have a look.